### PR TITLE
allow int and string versions

### LINKED ISSUE #14: Allow strings for version fields
It's not always convenient to use plain ints for versions.

### PR DESCRIPTION
**Pull request by tsnobip**
supersedes #21 
fixes #14

### CODE REVIEW

File: lib/generator.ml [hunk 1]
```diff
@@ -101,50 +98,71 @@ let make_get_version_from_json = function
...
-  | _ ->
-    invalid_arg "The parsed JSON should be an object."
+| `Assoc fields ->
+  let version =
+    List.find_map
+      ~f:(function
+        | "version", `String version ->
+          Some (`String version)
+        | "version", `Int version ->
+          Some (`Int version)
```
Comment (wokalski) [inline lib/generator.ml:108]:
```suggestion
        | "version", (`String _ | `Int _) as version ->
          Some version
```

File: lib/generator.ml [hunk 2]
```diff
@@ -301,27 +332,33 @@ let make_upgraders ?(impl_kind = Config.Native) ?output_prefix = function
...
-            let _sorted_items, _type_map, main_type, _main_type_param_count =
-              Upgrader.load_sort_map (recreate_path newest_version)
-            in
-            main_type
+        let* ( _sorted_items
+             , _type_map
+             , main_type
+             , _main_type_param_count
+             , latest_version )
+          =
```
Comment (wokalski) [inline lib/generator.ml:351]: This generally points in the direction of creating separate functions to just return those few fields specifically. The exception being if you need to compute sorted_items/type_map/main_type_param_count anyway.

File: lib/upgrader.mli [hunk 3]
```diff
@@ -22,21 +28,28 @@ val user_fns_module_name : string -> string
...
+         * Ast.NoLoc.TypeExpr.t)
+         Stdlib.Map.Make(Stdlib.StringLabels).t
+       * string
+       * int
+       * Version.t
+     , [> `Empty_atd_file of string
+       | `Incoherent_version_field of string * string
+       | `No_version_field of string
+       ] )
+     result
```
Comment (wokalski) [inline lib/upgrader.mli:55]: This begs for some type aliases to make it more readable.

**Review by wokalski — CHANGES_REQUESTED** [review]
The changes look great. I think snapshots are not updated so it's hard to judge the changes to the generated code 😄

**Comment by wokalski** [comment]
The snapshots are updated, pardon. I suggest to change the snapshot tests to test for the int -> string case.

File: lib/upgrader.mli (hunk 3 above)
Comment (tsnobip) [reply lib/upgrader.mli:55]: agreed!

File: lib/generator.ml (hunk 2 above)
Comment (tsnobip) [reply lib/generator.ml:375]: yeah I know it looks ugly, but in practice for this case, the separate function would be pretty similar so I think it makes sense to keep it that way.

**Review by wokalski — APPROVED** [review]
Looks good!

**Pull request merged**